scripts/changesummary.py

```diff
@@ -0,0 +1,540 @@
+                artifacts.
+        """
+        if not os.path.exists(directory):
+            raise ArtifactsDirDoesNotExist(
+                "Artifacts directory does not " " exist : {0}".format(directory)
+            )
+
+    def _rewrite_as_sorted_json(self, file_path):
+        """Read the Json file at the specifed path, sort the data,
+            and write it back to file.
+
+        args:
+            file_path (str): The path to the json file to be sorted
+        """
+        with open(file_path, "r+") as f:
+            data = json.load(f)
+            # Reset position to the beginning
+            f.seek(0)
+            # Sort the json file and write to disk
+            json.dump(data, f, indent=4, sort_keys=True)
+
+            # Truncate anything left as it's not needed
+            f.truncate()
+
+    def _load_json_to_dataframe(self, file_path):
+        """Returns a pandas dataframe from the json file provided.
+
+        args:
+            file_path (str): The relative path to the discovery artifact to
+                parse.
+        """
+
+        # Create an empty dataframe as we will need to return it if the file
+        # doesn't exist
+        dataframe_doc = None
+
+        if os.path.exists(file_path):
+            # The discovery artifact json file isn't sorted. Let's sort it first
+            # to make it easier for maintainers to see only the relevant changes
+            # when dicovery artifacts are updated rather than the changes due to
+            # random key ordering.
+            self._rewrite_as_sorted_json(file_path)
+
+            with open(file_path, "r+") as f:
+                # Now load the json file into a pandas dataframe as a flat table
+                dataframe_doc = pd.json_normalize(json.load(f))
+        return dataframe_doc
+
+    def _get_discovery_differences(self, filename):
+        """Returns a pandas dataframe which contains the differences with the
+        current and new discovery artifact directories, corresponding to the
+        file name provided.
+
+        args:
+            filename (str): The name of the discovery artifact to parse.
+        """
+        # The paths of the 2 discovery artifacts to compare
+        current_artifact_path = os.path.join(self._current_artifacts_dir, filename)
+        new_artifact_path = os.path.join(self._new_artifacts_dir, filename)
+
+        # Use a helper functions to load the discovery artifacts into pandas
+        # dataframes
+        current_doc = self._load_json_to_dataframe(current_artifact_path)
+        new_doc = self._load_json_to_dataframe(new_artifact_path)
+
+        # Concatenate the 2 dataframes, transpose them, and create
+        # a new dataframe called combined_docs with columns
+        # `Keys`, `CurrentValue`, `NewValue`.
+        combined_docs = (
+            pd.concat([current_doc, new_doc], keys=["CurrentValue", "NewValue"])
+            .transpose()
+            .rename_axis("Key")
```

Comment:
Looks like the recommended approach is to use keyword arguments rather than rely on "index" being the default axis

```suggestion
            .rename_axis(index="Key")
```

> We highly recommend using keyword arguments to clarify your intent.

https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.rename_axis.html